app/features/workout/bottom_bar/WorkoutBottomBar.js

```diff
@@ -5,14 +5,27 @@ import {
     View,
     Text,
     StyleSheet,
-    TouchableOpacity
+    TouchableOpacity,
+    Alert
 } from 'react-native';
 import ConnectedDeviceStatusScreen from './device_status/ConnectedDeviceStatusScreen';
 
 class WorkoutBottomBar extends Component {
     
     _onPressEndWorkout() {
-        this.props.endWorkout();
+        if (this.props.isLoggedIn) {
+            this.props.endWorkout();
+        } else {
+            Alert.alert(
+                'Heads up!',
+                "You are not logged in, the data from this workout will be lost,\nPlease sign in under settings to save your data",
+                [
+                  {text: 'Yes', onPress: () => this.props.endWorkout()},
+                  {text: 'No', style: 'cancel'},,
+                ],
+                { cancelable: false }
+              )            
+        }
```

Comment:
move this logic somewhere else, either the action creators or a saga, whatever makes more sense